### ISSUE TITLE
Implement Continuous Delivery to eliminate manual effort and get code released quicker

### ISSUE THREAD

**Issue by garethbowen** [medic/cht-core#9367]
**Describe the issue**

Currently releasing takes 2-3 days of manual effort, which costs in productivity. It also means there's a limit on how quickly and frequently releases are done, so in some cases code is approved and merged but projects still have to wait before it can be deployed into production.  It also means that Medic is the gatekeeper on when releases are made available for community contributed code.

**Describe the improvement you'd like**

Use Continuous Delivery to automatically release every time a non-chore commit is merged. This should work much like cht-conf which automatically releases using semantic-release to npm. This will include...

- updating the semver version
- publishing the artifacts
- generating and publishing the release notes
- sending appropriate notifications

**Describe alternatives you've considered**

- Doing releases _less_ often - this would reduce the workload but increase the time partners have to wait for changes that are ready

**Steps to achieve this**

Preparation steps - these can be done while still using the manual process.

- [ ] Automatically generate release notes. This might include the PR contributor including release note details (like highlights, screenshots, links to documentation, etc) in the issue or PR so that the script can publish these during release.
- [ ] Updating the release notes so you can aggregate changes between two versions. Ideally find a 3rd party solution for this, but implementers need to be able to see all the changes that will be applied from their current version to the desired upgrade version. This is really useful today, but will be critical when we increase the number of releases drastically. The old release notes will need to be updated in this new format so it still works if comparing back to 3.10 (approx). Releases notes earlier than that can just be archived.
- [ ] Improve how we label regressions and known issues. Right now we label every regression with an "Affects" label for every release between when it was introduced and when it was resolved. This is already unwieldy but will become more so with more releases. Mostly this label is used so the release notes can include "known issues" but searching on GH. With the new release notes format (above) this can be replaced with a better system, for example, just label when the regression was introduced, and the release notes can scrape all labels.
- [ ] Re-evaluate our supported versions policy. It was designed for when releases happened infrequently. With releases happening all the time does this change the support duration? How will fixes be backported to dozens of old branches anyway? If we guarantee no backwards compatibility issues, is it reasonable to expect people to upgrade to the latest minor?
- [ ] Announcing the release on the forum. The forum posts are very formulaic, mostly just linking to the release notes. Set up GH > Forum integrations and it should be straightforward.

Finally, in one go...

- [ ] Use semantic-release to automate kicking off the release
- [ ] Update documentation, issue templates, etc
- [ ] Communicate new process and policies on the forum, with teammates, and with key partners

**Comment by mrjones-plip** [comment]
Looking at [4.13 coming up](https://github.com/medic/cht-core/milestone/136?closed=1) with no features per se, it reminded me we likely should be sure to NOT do a CD release if there's only e2e or unit tests (or possibly only [technical](https://github.com/medic/cht-core/issues?q=%20is%3Aissue%20label%3A%22Type%3A%20Technical%20issue%22%20) tickets?) in the release.  

If we didn't filter on this,  3 new e2e tests that work as soon as they hit `main`, but have literally have no discernible user facing features in a release.

**Comment by garethbowen** [comment]
@mrjones-plip Agreed. These changes should be committed with the commit message starting with "chore" which in a CD world (eg: cht-conf) should not trigger a release.

**Comment by andrablaj** [comment]
After careful discussions, we have decided to focus our development efforts on features specifically requested by the community and those that CHT deployments will actively upgrade to adopt.

Given the nature of CHT Core deployments and our community's upgrade patterns, continuous delivery isn't practical for CHT Core at this time. We are closing this issue to keep our backlog focused on initiatives that will have the most direct impact on the community.

If circumstances change in the future, we can always revisit this topic.
